comm.go

```diff
@@ -41,6 +41,17 @@ func (p *PubSub) getHelloPacket() *RPC {
 }
 
 func (p *PubSub) handleNewStream(s network.Stream) {
+	peer := s.Conn().RemotePeer()
+
+	p.inboundStreamsMx.Lock()
+	other, dup := p.inboundStreams[peer]
+	if dup {
+		log.Debugf("duplicate inbound stream from %s; resetting other stream", peer)
+		other.Reset()
```

Comment:
Does the muxer guarantee that an outstanding read from this stream will error out immediately as soon as we reset it? Can't remember.

Reply:
I don't really know, but it shouldn't matter much. As long as it happens promptly enough we should be fine.